pkg/services/govmomi/service.go

```diff
@@ -118,58 +119,58 @@ func (vms *VMService) ReconcileVM(ctx *context.VMContext) (vm infrav1.VirtualMac
 	//
 
 	// Create a new virtualMachineContext to reconcile the VM.
-	vmCtx := &virtualMachineContext{
-		VMContext: *ctx,
-		Obj:       object.NewVirtualMachine(ctx.Session.Client.Client, vmRef),
+	virtualMachineCtx := &virtualMachineContext{
+		VMContext: *vmCtx,
+		Obj:       object.NewVirtualMachine(vmCtx.Session.Client.Client, vmRef),
 		Ref:       vmRef,
 		State:     &vm,
 	}
 	vm.VMRef = vmRef.String()
 
-	vms.reconcileUUID(vmCtx)
+	vms.reconcileUUID(virtualMachineCtx)
```

Comment:
```suggestion
	vms.reconcileUUID(ctx, virtualMachineCtx)
```
need to refactor function `reconcileUUID`

Please also check the following vms.reconcileXXX if they need refactoring